ci/vm/entrypoint.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+set -Exeo pipefail
+
+OS=$1
+BUILD_SOURCEVERSION=$2
+
+install_packages() {
+  if [[ $OS == *"ubuntu"* ]]; then
+    apt-get update &&
+      DEBIAN_FRONTEND=noninteractive apt-get install -yq \
+        automake \
+        dkms \
+        git \
+        libtool &&
+      apt-get clean && rm -rf /var/lib/apt/lists/*
+  elif [[ $OS == *"centos"* ]]; then
+    yum install -y -q centos-release-scl
+    yum install -y -q \
+      automake \
+      devtoolset-8-gcc \
+      devtoolset-8-gcc-c++ \
+      elfutils-libelf-devel \
+      git \
+      libtool \
+      make &&
+      yum clean all
+  fi
+}
+
+xpmem_build() {
+  git clone -q https://github.com/openucx/xpmem.git
+  cd xpmem
+  git checkout "$BUILD_SOURCEVERSION"
```

Comment:
Most likely PR branch is not in the default clone refspec, so such checkout may fail

Reply:
This is a built-in Azure var, that should refer to the PR branch.

Reply:
True, It requires a fetch by PR:
  git fetch origin pull/"$PR_NUM"/merge
  git checkout FETCH_HEAD

Reply:
Thanks again for caching this absolutely crucial point!